src/main/java/com/conveyal/datatools/editor/models/transit/TripPattern.java

```diff
@@ -48,57 +49,73 @@
 
-    public int getNumberOfTrips () {
-        FeedTx tx = VersionedDataStore.getFeedTx(this.feedId);
-        Collection<Trip> trips = tx.getTripsByPattern(this.id);
-        return trips == null ? 0 : trips.size();
+    // give the UI a little information about the content of this trip pattern
+    public transient Long numberOfTrips;
+    public transient Map<String, Long> tripCountByCalendar;
+
+
+    @JsonProperty("numberOfTrips")
+    public Long jsonGetNumberOfTrips () {
+        return numberOfTrips;
     }
 
-    public Map<String, Long> getTripCountByCalendar () {
-        FeedTx tx = VersionedDataStore.getFeedTx(this.feedId);
+    @JsonProperty("tripCountByCalendar")
+    Map<String, Long> jsonGetTripCountByCalendar () { return tripCountByCalendar; }
+
+    // do-nothing setters
+    @JsonProperty("numberOfTrips")
+    public void jsonSetNumberOfTrips(Long numberOfTrips) { }
+
+    @JsonProperty("tripCountByCalendar")
+    public void jsonSetTripCountByCalendar(Map<String, Long> tripCountByCalendar) { }
+
+    /** add transient info for UI with number of routes, number of trips */
+    public void addDerivedInfo(final FeedTx tx) {
         Collection<Trip> trips = tx.getTripsByPattern(this.id);
-        return trips.stream()
-                .filter(t -> t.calendarId != null)
+        numberOfTrips = Long.valueOf(trips.size());
+        tripCountByCalendar = trips.stream()
+                .filter(t -> t != null && t.calendarId != null)
                 .collect(Collectors.groupingBy(t -> t.calendarId, Collectors.counting()));
     }
 
     /**
      * Lines showing how stops are being snapped to the shape.
      * @return
      */
-    @JsonProperty("stopConnections")
-    public LineString[] jsonGetStopConnections () {
-        if (useStraightLineDistances || shape == null)
-            return null;
-
-        FeedTx tx = VersionedDataStore.getFeedTx(this.feedId);
-
-        try {
-            LineString[] ret = new LineString[patternStops.size()];
-
-            double[] coordDistances = getCoordDistances(shape);
-            LocationIndexedLine shapeIdx = new LocationIndexedLine(shape);
-
-            for (int i = 0; i < ret.length; i++) {
-                TripPatternStop ps = patternStops.get(i);
-
-                if (ps.shapeDistTraveled == null) {
-                    return null;
-                }
-
-                Coordinate snapped = shapeIdx.extractPoint(getLoc(coordDistances, ps.shapeDistTraveled));
-                // offset it slightly so that line creation does not fail if the stop is coincident
-                snapped.x = snapped.x - 0.00000001;
-                Coordinate stop = tx.stops.get(patternStops.get(i).stopId).location.getCoordinate();
-                ret[i] = GeoUtils.geometyFactory.createLineString(new Coordinate[] {stop, snapped});
-            }
-
-            return ret;
-        } finally {
-            tx.rollback();
-        }
-
-    }
+//    @JsonProperty("stopConnections")
```

Comment:
Man, do we not show the little lines connecting stops to shapes in the UI anymore? :cry: